ourhexenv.py

```diff
@@ -55,6 +56,13 @@ def __init__(self, board_size=5, sparse_flag=True):
         self.PLAYER2 = (50, 50, 255)    # Blue
         self.EMPTY = (255, 255, 255)    # White
 
+        # Union Find Check Winner Setup
+        self.uf = UnionFind(board_size * board_size + 4)  # Extra 4 for virtual nodes
```

Comment:
Now that we are using a helper object to track the state of the board, we have to do this initialization on the reset step

Reply:
Yes! Great catch!